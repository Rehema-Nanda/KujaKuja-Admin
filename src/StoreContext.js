// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
