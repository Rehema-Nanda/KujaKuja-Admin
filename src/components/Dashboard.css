
.dashboard{
    padding-top: 20px;
    min-height: calc(100vh - 248px);
}

.dashboard .login-bar{
    margin-left: 4px;
    margin-right: 22px;
    margin-bottom: -64px;
    padding: 0px;
    display: flex;
    flex-direction: row;
}

.dashboard .login-bar .welcome{
    flex-basis: 50%;
}

.dashboard .login-bar .last-login{
    flex-basis: 50%;
    text-align: right;
}

.dashboard .top-container{
    background-color: #FFFFFF;
    border-radius: 8px;
    margin-top: 48px;
    box-shadow: 0 24px 18px -10px rgba(23,29,33,0.08);
    margin-bottom: 56px;
    width: 100%;
    padding: 16px;
}

.dashboard .data-holder{
    height: 136px;
    padding-top: 48px;
}

.dashboard .top-details-row{
    margin-top: 24px;
    margin-bottom: 64px;
}

.dashboard .pie{
    display: flex;
    flex-direction: row;
}

.dashboard .pie .pie-holder{
    height: 136px;
    flex-basis: 38%;
    padding: 8px;
}

.dashboard .pie .pie-legend{
    flex-basis: 62%;
    padding: 24px;
}

.top-container-idea-feed{
    overflow-y: auto;
    height: 640px;
    background-color: #FFFFFF;
    border-radius: 8px;
    margin-top: 48px;
    box-shadow: 0 24px 18px -10px rgba(23,29,33,0.08);
    margin-bottom: 56px;
    width: 100%;
    padding: 16px;
}

.top-container-idea-feed li{
    padding: 12px;
}

.top-container-idea-feed li:nth-child(even) {
    background: #FBFBFB;
    border-radius: 4px;

}

.top-container-idea-feed li .small{
    margin-top: -8px;
    margin-bottom: -0px;
}

.response-line {
    height: 48px;
}

.response-title {
    text-align: right;
}

td{
   word-wrap: break-word;
}
