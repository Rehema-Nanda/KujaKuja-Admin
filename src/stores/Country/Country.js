import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";

class Country {
    store = null;
    id = null;
    enabled = null;
    name = null;
    isoTwoLetterCode = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            enabled: this.enabled,
            name: this.name,
            iso_two_letter_code: this.isoTwoLetterCode,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
        };
    }

    updateFromJson(json) {
        this.enabled = json.enabled;
        this.name = json.name;
        this.isoTwoLetterCode = json.iso_two_letter_code;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteCountry(this.id);
        await this.store.removeCountry(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateCountry(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createCountry(json);
            this.id = responseData.id;
            await this.store.addCountry(this);
        }
    }
}
decorate(Country, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Country;
