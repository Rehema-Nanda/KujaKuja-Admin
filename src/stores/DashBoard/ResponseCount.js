export default class ResponseCount {
    store = null;
    satisfied = null;
    count = null

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.updateFromJson(json);
        }
    }

    updateFromJson(json) {
        this.satisfied = json.satisfied;
        this.count = json.cnt;
    }
}
