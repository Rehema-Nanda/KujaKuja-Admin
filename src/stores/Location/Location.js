import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";
import circle from "@turf/circle";
import _ from "lodash";

class Location {
    store = null;
    id = null;
    name = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;
    countryId = null;
    countryName = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            name: this.name,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            country_id: this.countryId,
        };
    }

    updateFromJson(json) {
        this.name = json.name;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.countryId = json.country_id;
        this.countryName = json.country_name;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && typeof this.geojson === "string") {
            this.geojson = JSON.parse(this.geojson);
        }
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }

        // TODO: https://trello.com/c/9pBBU3hU/962-some-locations-are-missing-geometry
        // if no geometry, draw a 5km radius circle around coords
        if (this.geojson && _.isEmpty(this.geojson.geometry)) {
            const locationCoords = [this.lng, this.lat];
            const radius = 5;
            const options = { units: "kilometers" };
            this.geojson = circle(locationCoords, radius, options);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteLocation(this.id);
        await this.store.removeLocation(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateLocation(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createLocation(json);
            this.id = responseData.id;
            await this.store.addLocation(this);
        }
    }
}

decorate(Location, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Location;
