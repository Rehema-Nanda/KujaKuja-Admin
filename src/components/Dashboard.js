import React from "react";
import {
    Container, Row, Col, Badge,
} from "reactstrap";
import moment from "moment";
import { observer } from "mobx-react";
import { Pie } from "react-chartjs-2";
import formatNumber from "../helpers/formatNumber";
import { StoreContext } from "../StoreContext";
import "./Dashboard.css";
import Loading from "./Loading";

class Dashboard extends React.PureComponent {
    render() {
        const {
            locationStore, servicePointStore, serviceTypeStore, userStore, countryStore, dashboardStore: {
                responseCount, isLoading,
            }, featuredIdeaStore: { lastFeaturedIdea },
        } = this.context;

        const satisfiedResponsesCount = responseCount.length > 0 && parseInt(
            responseCount.filter((obj) => {
                return obj.satisfied;
            })[0].count,
            10,
        );

        const unsatisfiedResponsesCount = responseCount.length > 0 && parseInt(
            responseCount.filter((obj) => {
                return !obj.satisfied;
            })[0].count,
            10,
        );

        const totalResponsesCount = satisfiedResponsesCount + unsatisfiedResponsesCount;
        /* eslint-disable no-mixed-operators */
        const percentageSatisfied = (satisfiedResponsesCount / totalResponsesCount * 100).toFixed(1);
        const percentageUnsatisfied = (unsatisfiedResponsesCount / totalResponsesCount * 100).toFixed(1);
        const lastLogin = (userStore.userInfo.length > 0 && userStore.userInfo[0].lastSignInAt && moment(userStore.userInfo[0].lastSignInAt).format("h:mma MM/DD/YYYY")) || "";

        const chartData = {
            labels: [
                "Satisfied",
                "Unsatisfied",
            ],
            datasets: [
                {
                    data: [percentageSatisfied, percentageUnsatisfied],
                    backgroundColor: ["#00BA54", "#F8382B"],
                    hoverBackgroundColor: ["#00BA54", "#F8382B"],
                },
            ],
        };

        const chartOptions = {
            legend: {
                display: false,
            },
        };

        return (

            <div>
                <section className="dashboard">

                    <div className="title-background-parent">
                        <div className="title-background-shape" />
                    </div>

                    <Container>
                        <Row>
                            <Col>
                                <div className="login-bar">
                                    <div className="welcome">
                                        <p className="small">Welcome back!</p>
                                    </div>
                                    <div className="last-login">
                                        <p className="small">
                                            Last login:&nbsp;
                                            {lastLogin}
                                        </p>
                                    </div>
                                </div>
                            </Col>
                        </Row>

                        <Row>
                            <Col sm={{ size: 8 }}>
                                <Row>
                                    <Col>
                                        <div className="top-container">
                                            <h4 className="label">Total Data Points</h4>
                                            <div className="data-holder">
                                                <h2 className="numbers_big">{formatNumber(totalResponsesCount)}</h2>
                                            </div>
                                        </div>
                                    </Col>
                                    <Col>
                                        <div className="top-container">
                                            <h4 className="label">Overall Satisfaction</h4>
                                            <div className="pie">
                                                <div className="pie-holder">
                                                    <Pie
                                                        data={chartData}
                                                        options={chartOptions}
                                                        height={64}
                                                        width={64}
                                                    />
                                                </div>
                                                <div className="pie-legend">
                                                    <p className="small">
                                                        <Badge
                                                            color="#F8382B"
                                                            style={{ backgroundColor: "#F8382B" }}
                                                            pill
                                                        >
                                                            &nbsp;
                                                        </Badge>
                                                        &nbsp;
                                                        {`${percentageUnsatisfied}% Unsatisfied`}
                                                    </p>
                                                    <p className="small">
                                                        <Badge
                                                            color="success"
                                                            style={{ backgroundColor: "#00BA54" }}
                                                            pill
                                                        >
                                                            &nbsp;
                                                        </Badge>
                                                        &nbsp;
                                                        {`${percentageSatisfied}% Satisfied`}
                                                    </p>
                                                </div>
                                            </div>
                                        </div>
                                    </Col>
                                </Row>

                                <Row className="top-details-row">
                                    <Col sm={{ size: 4 }}>
                                        <div>
                                            <h4 className="label">Active Countries</h4>
                                            <h2 className="numbers_big left">{formatNumber(countryStore.enabledCountries.length)}</h2>
                                        </div>
                                    </Col>
                                    <Col sm={{ size: 4 }}>
                                        <div>
                                            <h4 className="label">Active Locations</h4>
                                            <h2 className="numbers_big left">{formatNumber(locationStore.locations.length)}</h2>
                                        </div>
                                    </Col>
                                    <Col sm={{ size: 4 }}>
                                        <div>
                                            <h4 className="label">Active Service Points</h4>
                                            <h2 className="numbers_big left">{formatNumber(servicePointStore.servicePoints.length)}</h2>
                                        </div>
                                    </Col>
                                </Row>

                                <Row>
                                    <Col sm={{ size: 4 }}>
                                        <div>
                                            <h4 className="label">Service Types</h4>
                                            <h2 className="numbers_big left">{formatNumber(serviceTypeStore.serviceTypes.length)}</h2>
                                        </div>
                                    </Col>
                                    <Col sm={{ size: 4 }}>
                                        <div>
                                            <h4 className="label">Team Members</h4>
                                            <h2 className="numbers_big left">{formatNumber(userStore.users.length)}</h2>
                                        </div>
                                    </Col>
                                </Row>
                            </Col>

                            <Col sm={{ size: 4 }}>
                                <Col>
                                    <div className="top-container-idea-feed">
                                        <h4 className="label">This Week&apos;s Featured Ideas</h4>
                                        <ul className="list-unstyled">
                                            {lastFeaturedIdea.length === 0
                                            && (
                                                <li>
                                                    <p>No featured ideas this week</p>
                                                </li>
                                            )}
                                            {lastFeaturedIdea && lastFeaturedIdea.map((row) => {
                                                return (
                                                    <li key={row.id}>
                                                        <p>{row.idea}</p>
                                                        <p className="small">
                                                            {moment(row.updated_at).format("L")}
                                                            &nbsp;|&nbsp;
                                                            {row.name}
                                                        </p>
                                                    </li>
                                                );
                                            })}
                                        </ul>
                                    </div>
                                </Col>
                            </Col>
                        </Row>
                    </Container>

                    {isLoading && <Loading /> }

                </section>
            </div>
        );
    }
}

Dashboard.contextType = StoreContext;

export default observer(Dashboard);
