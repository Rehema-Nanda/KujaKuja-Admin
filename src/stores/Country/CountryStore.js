import {
    flow, decorate, observable, computed,
} from "mobx";
import Country from "./Country";
import CountryApiService from "./CountryApiService";
import AppConfig from "../../AppConfig";

class CountryStore {
    countries = [];
    count = 0;
    enabledCountries = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new CountryApiService(AppConfig.API_URL);

        this.loadCountries();
        this.loadEnabledCountries();
    }

    get countriesSortedByName() {
        return this.countries.slice().sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadCountries = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchAllCountries(config);
            this.countries.replace(countriesJson.data.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.count = countriesJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadEnabledCountries = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchEnabledCountries();
            this.enabledCountries.replace(countriesJson.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load enabled countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateCountryFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const countriesJson = yield this.apiService.fetchCountryById(id);
        let country = this.countries.find((c) => {
            return c.id === id;
        });

        if (countriesJson.length === 0 && country) {
            // country no longer exists on the server, remove it locally
            yield this.removeCountry(country);
            country = null;
        }
        if (countriesJson.length > 0) {
            if (country) {
                country.updateFromJson(countriesJson[0]);
            }
            else {
                country = new Country(this, countriesJson[0]);
                this.countries.push(country);
            }
        }
        this.isLoading = false;
        return country;
    });

    removeCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.splice(this.countries.indexOf(country), 1);
    });

    addCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.push(country);
    });
}
decorate(CountryStore, {
    countries: observable,
    enabledCountries: observable,
    countriesSortedByName: computed,
    isLoading: observable,
});

export default CountryStore;
