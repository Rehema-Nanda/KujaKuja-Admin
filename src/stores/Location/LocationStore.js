import {
    flow, decorate, observable, computed,
} from "mobx";
import Location from "./Location";
import LocationApiService from "./LocationApiService";
import AppConfig from "../../AppConfig";

class LocationStore {
    locations = [];
    locationsSyndicated = [];
    isLoading = false;
    count = 0;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new LocationApiService(AppConfig.API_URL);

        this.loadLocations();
        this.loadSyndicatedLocations();
    }

    get locationsSortedByName() {
        return this.locations.sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadLocations = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchAllLocations(config);
            this.locations.replace(locationsJson.data.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.count = locationsJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadSyndicatedLocations = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchSyndicatedLocations();

            this.locationsSyndicated.replace(locationsJson.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load syndicated locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateLocationFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const locationsJson = yield this.apiService.fetchLocationById(id);
        let location = this.locations.find((loc) => {
            return loc.id === id;
        });

        if (locationsJson.length === 0 && location) {
            // location no longer exists on the server, remove it locally
            yield this.removeLocation(location);
            location = null;
        }
        if (locationsJson.length > 0) {
            if (location) {
                location.updateFromJson(locationsJson[0]);
            }
            else {
                location = new Location(this, locationsJson[0]);
                this.locations.push(location);
            }
        }
        this.isLoading = false;
        return location;
    });

    removeLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.splice(this.locations.indexOf(location), 1);
    });

    addLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.push(location);
    });
}
decorate(LocationStore, {
    locations: observable,
    locationsSyndicated: observable,
    locationsSortedByName: computed,
    isLoading: observable,
});

export default LocationStore;
