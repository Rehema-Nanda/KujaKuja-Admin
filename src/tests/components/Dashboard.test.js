import React from "react";
import { mount } from "enzyme";
import Dashboard from "../../components/Dashboard";
import { StoreProvider } from "../../StoreContext";
import CountryStore from "../../stores/Country/CountryStore";
import Country from "../../stores/Country/Country";
import LocationStore from "../../stores/Location/LocationStore";
import Location from "../../stores/Location/Location";
import UserStore from "../../stores/User/UserStore";
import User from "../../stores/User/User";
import ServiceTypeStore from "../../stores/ServiceType/ServiceTypeStore";
import ServiceType from "../../stores/ServiceType/ServiceType";
import ServicePoint from "../../stores/ServicePoint/ServicePoint";
import ServicePointStore from "../../stores/ServicePoint/ServicePointStore";
import DashboardStore from "../../stores/DashBoard/DashBoardStore";
import ResponseCount from "../../stores/DashBoard/ResponseCount";

jest.mock("../../stores/Country/CountryStore");
jest.mock("../../stores/Location/LocationStore");
jest.mock("../../stores/User/UserStore");
jest.mock("../../stores/ServiceType/ServiceTypeStore");
jest.mock("../../stores/ServicePoint/ServicePointStore");
jest.mock("../../stores/DashBoard/DashBoardStore");

ServiceTypeStore.mockImplementation(() => {
    return {
        serviceTypes: [
            new ServiceType(
                null,
                {
                    id: "1",
                    name: "Test Service Type",
                    created_at: null,
                    updated_at: null,
                },
            ),
        ],
    };
});

CountryStore.mockImplementation(() => {
    return {
        enabledCountries: [
            new Country(
                null,
                {
                    id: "1",
                    enabled: false,
                    name: "Afghanistan",
                    iso_two_letter_code: "AF",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                },
            ),
        ],
    };
});

LocationStore.mockImplementation(() => {
    return {
        locations: [
            new Location(
                null,
                {
                    id: "1",
                    name: "Test Location",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                    country_id: "1",
                    country_name: "Afghanistan",
                },
            ),
        ],
    };
});

UserStore.mockImplementation(() => {
    return {
        userInfo: [
            new User(
                null,
                {
                    id: "1",
                    email: "testuser@kujakuja.com",
                    encrypted_password: "password",
                    current_sign_in_at: "2020-01-01T00:00:00.000Z",
                    created_at: null,
                    updated_at: null,
                    settlement_id: "1",
                    settlement_name: "Test Location",
                    is_admin: true,
                    is_survey: false,
                    is_service_provider: false,
                },
            ),
        ],
        users: [
            new User(
                null,
                {
                    id: "1",
                    email: "testuser@kujakuja.com",
                    encrypted_password: "password",
                    current_sign_in_at: "2020-01-01T00:00:00.000Z",
                    created_at: null,
                    updated_at: null,
                    settlement_id: "1",
                    settlement_name: "Test Location",
                    is_admin: true,
                    is_survey: false,
                    is_service_provider: false,
                },
            ),
        ],
    };
});

ServicePointStore.mockImplementation(() => {
    return {
        servicePoints: [
            new ServicePoint(
                null,
                {
                    id: "1",
                    name: "Test Service Point",
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                    service_type_id: "1",
                    service_type_name: "Test Service Type",
                    settlement_id: "1",
                    settlement_name: "Test Location",
                },
            ),
        ],
    };
});

DashboardStore.mockImplementation(() => {
    return {
        isLoading: false,
        responseCount: [
            new ResponseCount(
                null,
                {
                    satisfied: false,
                    cnt: "5",
                },
            ),
            new ResponseCount(
                null,
                {
                    satisfied: true,
                    cnt: "100",
                },
            ),
        ],
    };
});

describe("<Dashboard />", () => {
    const wrapper = mount(<Dashboard />, { wrappingComponent: StoreProvider });

    it("renders correctly", () => {
        // set last sign to prevent snapshot fail
        wrapper.setState({
            userProfile: { last_sign_in_at: "2019-08-15" },
        });

        expect(wrapper.debug()).toMatchSnapshot();
    });
});
