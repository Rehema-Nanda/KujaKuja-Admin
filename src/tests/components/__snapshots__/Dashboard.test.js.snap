// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
"<Dashboard>
  <div>
    <section className=\\"dashboard\\">
      <div className=\\"title-background-parent\\">
        <div className=\\"title-background-shape\\" />
      </div>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col tag=\\"div\\" widths={{...}}>
                <div className=\\"col\\">
                  <div className=\\"login-bar\\">
                    <div className=\\"welcome\\">
                      <p className=\\"small\\">
                        Welcome back!
                      </p>
                    </div>
                    <div className=\\"last-login\\">
                      <p className=\\"small\\">
                        Last login: 
                      </p>
                    </div>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-8\\">
                  <Row tag=\\"div\\">
                    <div className=\\"row\\">
                      <Col tag=\\"div\\" widths={{...}}>
                        <div className=\\"col\\">
                          <div className=\\"top-container\\">
                            <h4 className=\\"label\\">
                              Total Data Points
                            </h4>
                            <div className=\\"data-holder\\">
                              <h2 className=\\"numbers_big\\">
                                105
                              </h2>
                            </div>
                          </div>
                        </div>
                      </Col>
                      <Col tag=\\"div\\" widths={{...}}>
                        <div className=\\"col\\">
                          <div className=\\"top-container\\">
                            <h4 className=\\"label\\">
                              Overall Satisfaction
                            </h4>
                            <div className=\\"pie\\">
                              <div className=\\"pie-holder\\">
                                <Pie data={{...}} options={{...}} height={64} width={64}>
                                  <ChartComponent data={{...}} options={{...}} height={64} width={64} type=\\"pie\\" legend={{...}} redraw={false} datasetKeyProvider={[Function]}>
                                    <canvas height={64} width={64} id={[undefined]} onClick={[Function]} />
                                  </ChartComponent>
                                </Pie>
                              </div>
                              <div className=\\"pie-legend\\">
                                <p className=\\"small\\">
                                  <Badge color=\\"#F8382B\\" style={{...}} pill={true} tag=\\"span\\">
                                    <span style={{...}} className=\\"badge badge-#F8382B badge-pill\\">
                                       
                                    </span>
                                  </Badge>
                                   
                                  4.8% Unsatisfied
                                </p>
                                <p className=\\"small\\">
                                  <Badge color=\\"success\\" style={{...}} pill={true} tag=\\"span\\">
                                    <span style={{...}} className=\\"badge badge-success badge-pill\\">
                                       
                                    </span>
                                  </Badge>
                                   
                                  95.2% Satisfied
                                </p>
                              </div>
                            </div>
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                  <Row className=\\"top-details-row\\" tag=\\"div\\">
                    <div className=\\"top-details-row row\\">
                      <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                        <div className=\\"col-sm-4\\">
                          <div>
                            <h4 className=\\"label\\">
                              Active Countries
                            </h4>
                            <h2 className=\\"numbers_big left\\">
                              1
                            </h2>
                          </div>
                        </div>
                      </Col>
                      <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                        <div className=\\"col-sm-4\\">
                          <div>
                            <h4 className=\\"label\\">
                              Active Locations
                            </h4>
                            <h2 className=\\"numbers_big left\\">
                              1
                            </h2>
                          </div>
                        </div>
                      </Col>
                      <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                        <div className=\\"col-sm-4\\">
                          <div>
                            <h4 className=\\"label\\">
                              Active Service Points
                            </h4>
                            <h2 className=\\"numbers_big left\\">
                              1
                            </h2>
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                  <Row tag=\\"div\\">
                    <div className=\\"row\\">
                      <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                        <div className=\\"col-sm-4\\">
                          <div>
                            <h4 className=\\"label\\">
                              Service Types
                            </h4>
                            <h2 className=\\"numbers_big left\\">
                              1
                            </h2>
                          </div>
                        </div>
                      </Col>
                      <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                        <div className=\\"col-sm-4\\">
                          <div>
                            <h4 className=\\"label\\">
                              Team Members
                            </h4>
                            <h2 className=\\"numbers_big left\\">
                              1
                            </h2>
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </div>
              </Col>
              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-4\\">
                  <Col tag=\\"div\\" widths={{...}}>
                    <div className=\\"col\\">
                      <div className=\\"top-container-idea-feed\\">
                        <h4 className=\\"label\\">
                          This Week&#39;s Featured Ideas
                        </h4>
                        <ul className=\\"list-unstyled\\">
                          <li>
                            <p>
                              No featured ideas this week
                            </p>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </Col>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </section>
  </div>
</Dashboard>"
`;
